sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#sigilrecorder
+=end
+
+custom_require.call(%w[common common-arcana common-items common-travel drinfomon events spellmonitor equipmanager])
+
+class SigilRecorder
+	require 'yaml'
+
+	def initialize
+		#initialize arrays for supported towns, sigils, and seasons
+		@city_list = ['Crossing', 'Riverhaven', 'Shard']
+		@sigil_list = ['abolition','congruence','induction','permutation','rarefaction', 'antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+		@season_list = ['summer', 'fall', 'winter', 'spring']
+
+		#pull data and settings and store the current region and season
+		@data = get_data('sigils')
+		@settings = get_settings
+		@region = get_region #parse Room.current.image for which region/city we're in
+		@season = get_season #make sure we're recording sigils to the correct season
+		
+		#generate an array of all the room numbers associated with the current room's Room.current.image value
+		region_list = region_room_list
+		#subtract known hunting spots and any other rooms in the YAML's nogo list to avoid combat and rooms with known mapping issues
+		walk_list = generate_nogo_list(region_list)
+		
+		#debug info
+		DRC.message("Region detected: #{@region}")
+		DRC.message("Season detected: #{@season.capitalize}")
+
+		#perform any outdoorsmanship buffs as this is the skill that governs sigil hunting
+		DRC.message("Checking for \"outdoors\" waggle set...")
+		DRCA.do_buffs(@settings, 'outdoors') #perc sigil uses Outdoorsmanship
+		
+		#flag known patterns that suggest the room is unsuitable for sigil hunting or harvesting
+		Flags.add('sigil-danger-room', /.*begins to advance on you.*|You weave back and forth in a dance-like motion|You must be closer to use tactical abilities on your opponent/)
+
+		#main travel loop, travels to rooms and then kicks off the sigil hunting process
+		walk_list.each do |roomid|
+			
+			#check if the room title is in the 'banpatterns' array before travelling there. Skip and add to nogo if a match is found
+			banflag = false
+			@data.SigilInfo['banpatterns'].each do |pattern|
+				if Map[roomid].title.to_s.include? pattern
+					add_to_nogo(roomid, @data)
```

Comment:
🧹 The method `add_to_nogo` is always called with the same second argument, `@data`. Thoughts on removing the second argument from the method calls and just reference it in the method itself?

Reply:
I did this to avoid modifying instance variables outside the initialize function. The function takes in @data, modifies it, then returns it, so that @data is only modified in initialize.

This does mean that in record_sigil the YAML file is more up-to-date than the data in @data but the out of date portions are the portions the script has already gone over. But I'd be worried about changing the size of arrays/hashes in @data while other parts of the script are looping through indices/keys.